nshm_logic_tree_utilities/lib/loading_functions.py

```diff
@@ -0,0 +1,293 @@
+    """
+    Load aggregate statistics for all logic trees in the results_directory.
+
+    Parameters
+    ----------
+    results_directory : Union[Path, str]
+        The directory containing the results of one or more logic trees.
+        Should contain directories named like logic_tree_index_0, logic_tree_index_0 etc.
+    locations : tuple[str], optional
+        The locations to plot. Default is ("AKL", "WLG", "CHC").
+
+    Returns
+    -------
+    LoadedResults
+        An instance of LoadedResults containing the data and run notes for all logic trees in the directory.
+    """
+
+    config = cfg.Config()
+
+    if isinstance(results_directory, str):
+        results_directory = Path(results_directory)
+
+    results_df = pd.DataFrame()
+
+    for run_dir in results_directory.iterdir():
+        if run_dir.is_dir():
+            results_df = pd.concat(
+                [
+                    results_df,
+                    load_aggregate_stats_for_one_logic_tree_several_locations(
+                        run_dir, locations
+                    ),
+                ],
+                ignore_index=True,
+            )
+
+    return LoadedResults(
+        data_df=results_df,
+        collated_notes_df=pd.read_csv(
+            results_directory / config.get_value("collated_notes_file_name")
+        ),
+    )
+
+
+def insert_ln_std(data_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    Insert the natural logarithm of the standard deviation (std_ln) into the DataFrame.
+
+    This function iterates over the rows of the input DataFrame, checks for rows where the 'agg' column is 'cov',
+    calculates the std_ln, and appends these new rows to the DataFrame.
+
+    Parameters
+    ----------
+    data_df : pd.DataFrame
+        The input DataFrame containing the data.
+
+    Returns
+    -------
+    pd.DataFrame
+        A new DataFrame with the std_ln rows added.
+    """
+
+    # Initialize an empty DataFrame to hold the new rows
+    new_rows = []
+
+    # Iterate over the DataFrame
+    for index, row in data_df.iterrows():
+        # Append the current row to the list of new rows
+        new_rows.append(row)
+
+        # Check if the 'agg' column is 'cov'
+        if row["agg"] == "cov":
+
+            cov_arr = row.loc["values"]
+            std_ln_arr = np.sqrt(np.log(cov_arr**2 + 1))
+
+            # Create a new row that's a copy of the current row
+            new_row = row.copy()
+
+            # Update the 'agg' and 'values' columns of the new row
+            new_row["agg"] = "std_ln"
+            new_row["values"] = std_ln_arr
+
+            # Append the new row to the list of new rows
+            new_rows.append(new_row)
+
+    # Create a new DataFrame from the list of new rows
+    new_df = pd.DataFrame(new_rows)
+
+    return new_df
+
+
+def lookup_realization_name_from_hash(
+    individual_realization_df: pd.DataFrame, registry_directory: Union[Path, str]
+) -> list[plotting_utilities.RealizationName]:
```

Comment:
That's a good point but I think I will leave it as this spelling because I think I consistently used this spelling so if I try to change them all, I might miss some, and having inconsistent spelling might be worse